### PR TITLE
IPC: rpmsg: Changed checking endpoints on close

### PR DESCRIPTION
**Pull request by pawelosyp**
Initially this change was part of #56113 and there discussion about that started.
This change is due to situation I have got in project - network core creates two endpoints and app core creates only one. In that situation there are two endpoints, both of them have names and one of them is bounded - unused one has name.
So checking bounded by name returned false result in case of unused endpoint.

### CODE REVIEW

**Review by carlocaione — CHANGES_REQUESTED** [review]
You need to change https://github.com/zephyrproject-rtos/zephyr/blob/c2ec3d49cd2f952e0ead3dc7415d6cc9845a0db8/include/zephyr/ipc/ipc_service.h#L223-L225

Also please add an entry in the changelog file for this

File: include/zephyr/ipc/ipc_service.h [hunk 1]
```diff
@@ -220,9 +220,9 @@ int ipc_service_open_instance(const struct device *instance);
 
 /** @brief Close an instance
  *
- *  Function to be used to close an instance. All endpoints must be
+ *  Function to be used to close an instance. All bounded endpoints must be
  *  deregistered using ipc_service_deregister_endpoint before this
- *  is called.
+ *  is called (the unused one can be left).
```
Comment (carlocaione) [inline include/zephyr/ipc/ipc_service.h:225]:
```suggestion
 *  is called.
```

Comment (carlocaione) [reply include/zephyr/ipc/ipc_service.h:225]: Please add a note about this in `doc/releases/release-notes-3.4.rst`. It is an API change, so better document it.

**Review by carlocaione — CHANGES_REQUESTED** [review]

**Review by hubertmis — DISMISSED** [review]

File: doc/releases/release-notes-3.4.rst [hunk 2]
```diff
@@ -140,6 +140,9 @@ Changes in this release
   using picolibc, only applications using picolibc will be affected by this
   change at this time.
 
+* Changed implementation of checking for bounded endpoints during closing
+  an IPC instance. The new one checks by bounded flag.
```
Comment (carlocaione) [inline doc/releases/release-notes-3.4.rst:144]: This is in the wrong section. Please add a new `IPC` section under the `Subsystems` section and reword like:
```
:c:func:`ipc_service_close_instance` now only acts on bounded endpoints.
```

**Review by carlocaione — CHANGES_REQUESTED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by eobalski — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Pull request merged**